compiler/plugins/target/ROCM/Dialect/ROCM/Transforms/ApplyBuiltinPDLPatterns.cpp

```diff
@@ -232,16 +247,24 @@ class ApplyBuiltinPDLPatternsPass
     }
     if (enableTensorUKernels) {
       for (std::string target : targets) {
-        std::string builtinName =
-            llvm::formatv("ukernel_patterns_{}.mlir", target);
-        std::optional<StringRef> maybeBuiltin =
-            rocmDialect->getBuiltin(builtinName);
-        if (!maybeBuiltin) {
+        SmallVector<llvm::StringRef> allBuiltinNames =
+            rocmDialect->getBuiltinNames();
+        SmallVector<llvm::StringRef> builtinNames =
+            filterUkernelPatternsByTarget(allBuiltinNames, target);
+        std::string builtinSrc;
+        for (auto builtinName : builtinNames) {
+          std::optional<StringRef> maybeBuiltin =
+              rocmDialect->getBuiltin(builtinName);
+          if (maybeBuiltin) {
+            builtinSrc += maybeBuiltin.value().str() + "\n";
```

Comment:
Instead of concatenating the files, can't you just populate all files separately using `populatePDLModuleFromBuiltin` inside a loop?